Sources/macro.js

```diff
@@ -257,7 +257,7 @@ export function algo(publicAPI, model, numberOfInputs, numberOfOutputs) {
   }
 
   function getInputData(port = 0) {
-    if (!model.inputData[port]) {
+    if (!model.inputData[port] && model.inputConnection[port]) {
```

Comment:
Yes, because update should try to get the new output of its input. When you have static data, you already have it. If you have a connection, you should ask for it (always).

If the pipeline didn't change, then the ```model.inputConnection[port]()``` is almost a NoOp() as it directly return its output. But if the filter figure out that it has some changes, then it will recompute itself in order to give you the proper output.